### PR TITLE
gh-90868: Add _PyStaticObject_CheckRefcnt() function

### PR DESCRIPTION
**Pull request by vstinner**
Add _PyStaticObject_CheckRefcnt() function to make _PyStaticObjects_CheckRefcnt() shorter. Use
_PyObject_ASSERT_FAILED_MSG() to log the object causing the fatal error.

* Issue: gh-90868

### CODE REVIEW

**Comment by vstinner** [comment]
@ericsnowcurrently @kumaraditya303: This PR makes _PyStaticObjects_CheckRefcnt() more readable and shorter, what do you think?

File: Tools/build/generate_global_objects.py [hunk 1]
```diff
@@ -385,14 +385,22 @@ def generate_global_object_finalizers(immortal_objects):
...
+
+        printer.write("""
+static inline void
+_PyStaticObject_CheckRefcnt(PyObject *obj) {
+    if (Py_REFCNT(obj) < _PyObject_IMMORTAL_REFCNT) {
+        _PyObject_ASSERT_FAILED_MSG(obj,
+            "immortal object has less refcnt than expected "
+            "_PyObject_IMMORTAL_REFCNT");
+    }
+}
```
Comment (ericsnowcurrently) [inline Tools/build/generate_global_objects.py]: Rather that being part of the generated code, you could put this directly in pycore_global_objects_fini_generated.h, before the `/* The following is auto-generated by Tools/build/generate_global_objects.py. */` marker line.

**Review by ericsnowcurrently — CHANGES_REQUESTED** [review]
I have one suggestion.  Otherwise this looks good.

File: Tools/build/generate_global_objects.py (hunk 1 above)
Comment (vstinner) [reply Tools/build/generate_global_objects.py]: Aha, I didn't notice that only some parts of this file are generated. Ok, I added my new function at the top.

**Comment by vstinner** [comment]
I have made the requested changes; please review again.

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
Merged, thanks for the review.

**Pull request merged**